src/Carousel.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Carousel.css';
+
+function Carousel() {
+  let SliderArr = ['Первый', 'Второй', 'Последний'];
+
+  const [x, setX] = useState(0);
+  const [startX, setStartX] = useState(0);
+  const [offsetX, setOffsetX] = useState(0);
+
+  const [mouseDown, setMouseDown] = useState(false);
+
+  const widthOfTheSlide = 100;
+  const standardOfDifference = 50;
+
+  // Вот эта херня нужна чтобы включать transition только при смене слайда
+  const [timer, setTimer] = useState(0);
+  const [timerGo, setTimerGo] = useState(false);
+
+  const id = useRef(0);
+
+  const TIMER_TIME = 1; // время transition в секундах
+
+  const clear = () => {
+    window.clearInterval(id.current);
+  };
+
+  React.useEffect(() => {
+    id.current = window.setInterval(() => {
+      setTimer((time) => time - 1);
+    }, 1000);
+    return () => clear();
+  }, [timerGo]);
+
+  // чтобы таймер не ушел в минус
+  React.useEffect(() => {
+    if (timer <= 0  || timerGo === false) {
+      clear();
+      setTimerGo(false);
+    }
+  }, [timer, timerGo]);
+  // Эта самая херня для transition до сюда
+
+
+  function goLeft() {
+    x < 0 &&
```

Comment:
Is there any particular reason for using an if like this?
we're not in rendering, it's better to use the if () {}